src/Reflection/ReflectionMethod.php

```diff
@@ -294,4 +299,71 @@ public function getImplementingClass() : ReflectionClass
     {
         return $this->implementingClass;
     }
+
+    /**
+     * @param object|null $object
+     *
+     * @return \Closure
+     *
+     * @throws ClassDoesNotExist
+     * @throws NoObjectProvided
+     * @throws NotAnObject
+     * @throws ObjectNotInstanceOfClass
+     */
+    public function getClosure($object = null) : Closure
+    {
+        $declaringClassName = $this->getDeclaringClass()->getName();
+
+        if ($this->isStatic()) {
+            $this->assertClassExist($declaringClassName);
+
+            return function (...$args) use ($declaringClassName) {
+                return Closure::bind(function (string $declaringClassName, string $methodName, array $methodArgs) {
+                    return $declaringClassName::{$methodName}(...$methodArgs);
+                }, null, $declaringClassName)->__invoke($declaringClassName, $this->getName(), $args);
+            };
+        }
+
+        $this->assertObject($object);
+
+        return function (...$args) use ($declaringClassName, $object) {
+            return Closure::bind(function ($object, string $methodName, array $methodArgs) {
+                return $object->{$methodName}(...$methodArgs);
+            }, $object, $declaringClassName)->__invoke($object, $this->getName(), $args);
+        };
+    }
+
+    /**
+     * @throws ClassDoesNotExist
+     */
+    private function assertClassExist(string $className) : void
+    {
+        if ( ! \class_exists($className, false)) {
+            throw new ClassDoesNotExist('Method cannot be used as the class is not loaded');
```

Comment:
Please add the class name to the exception

Reply:
Done.